trace_decoder/src/processed_block_trace.rs

```diff
@@ -211,16 +202,16 @@ fn check_receipt_bytes(bytes: Vec<u8>) -> anyhow::Result<Vec<u8>> {
 /// Note that "*_accesses" includes writes.
 #[derive(Debug, Default)]
 pub(crate) struct NodesUsedByTxn {
-    pub state_accesses: Vec<H256>,
-    pub state_writes: Vec<(H256, StateWrite)>,
+    pub state_accesses: Vec<Address>,
+    pub state_writes: Vec<(Address, StateWrite)>,
 
-    // Note: All entries in `storage_writes` also appear in `storage_accesses`.
     pub storage_accesses: Vec<(H256, Vec<TrieKey>)>,
     #[allow(clippy::type_complexity)]
     pub storage_writes: Vec<(H256, Vec<(TrieKey, Vec<u8>)>)>,
+
     /// Hashed address -> storage root.
     pub accts_with_unaccessed_storage: HashMap<H256, H256>,
```

Comment:
> Any reason why the switch hasn't been done as well to accts_with_unaccessed_storage and self_destructed_accounts. We could use the new by_address methods for them too I believe

@Nashtare, I've moved `self_destructed_accounts`, which works, but having difficulty with `accts_with_unaccessed_storage`:
1. The key is `H256` because it comes from `all_accounts_in_pre_image`: https://github.com/0xPolygonZero/zk_evm/blob/9d7055dd0f04ca8dbacedf8a759e7b91a348fe59/trace_decoder/src/processed_block_trace.rs#L168-L176

2. Which ultimately comes from iteration of the input state trie: https://github.com/0xPolygonZero/zk_evm/blob/9d7055dd0f04ca8dbacedf8a759e7b91a348fe59/trace_decoder/src/lib.rs#L386-L391

I don't think we can iterate e.g `Address`es from a `HashedPartialTrie` because the `hash` is one-way - is that correct?

Reply:
Ah yes correct, we'll have to stick with H256